### PR TITLE
Terminate all slots in a set when clients are gone

### PR DESCRIPTION
**Pull request by hawkw**
Currently, the `SetWorker::run` function will terminate all per-slot tasks in the set when termination is explicitly signaled. However, it does *not* currently do this when the `SetWorker` shuts down because there are no more clients, so we don't clean up the per-slot tasks in that case. I noticed this [while reviewing][1] PR #45, but left the review only after it merged.

This commit fixes that by moving the termination code out of the loop, and making both the "explicit termination channel fired" and the "no more clients" cases `break` out of the loop into the termination code, instead of `return`ing.

[1]: https://github.com/oxidecomputer/qorb/pull/45#discussion_r1802145192

### CODE REVIEW

File: src/slot.rs [hunk 1]
```diff
@@ -855,29 +855,11 @@ impl<Conn: Connection> SetWorker<Conn> {
         loop {
             tokio::select! {
                 biased;
-                // If we should exit, terminate immediately
+                // If we should exit, terminate immediately.
```
Comment (hawkw) [inline src/slot.rs:858]: Whoops, I didn't need to add the period, my bad --- I can remove this if Git diff minimalism is important to you. :)

Comment (smklein) [reply src/slot.rs:858]: nah, don't worry about it

**Review by smklein — APPROVED** [review]
Thanks for patching this!

**Comment by smklein** [comment]
I'm gonna merge this so i can rebase https://github.com/oxidecomputer/qorb/pull/48 to use it

**Pull request merged**